### ISSUE TITLE
out of range exception reading UnicodeStorage during UpdatePatternLocations

### ISSUE THREAD

**Issue by DHowett** [microsoft/terminal#12837]
1.12.10908

Uncertain as to repro steps.

Watson failure hash 326e3a1b-889b-f24e-582d-0e18d3c853b4

<html>
<body>

Frame | Name | | offset | | | | file || line
-- | -- | -- | -- | -- | -- | -- | -- | -- | --
0 | KERNELBASE!RaiseException |   | 0x000000000000006C |   |   |   | xcpt.c |   | 936
1 | VCRUNTIME140!_CxxThrowException |   | 0x0000000000000090 |   |   |   | throw.cpp |   | 75
2 | msvcp140!std::_Xout_of_range |   | 0x0000000000000022 |   |   |   | xthrow.cpp |   | 25
3 | Microsoft_Terminal_Control!CharRow::GetText |   | 0x000000000004414C |   |   |   | CharRow.cpp |   | 217
4 | Microsoft_Terminal_Control!TextBuffer::GetPatterns |   | 0x00000000000000C0 |   |   |   | textBuffer.cpp |   | 2665
5 | Microsoft_Terminal_Control!Microsoft::Terminal::Core::Terminal::UpdatePatternsUnderLock |   | 0x000000000000008E |   |   |   | Terminal.cpp |   | 1255
6 | Microsoft_Terminal_Control!winrt::Microsoft::Terminal::Control::implementation::ControlCore::UpdatePatternLocations |   | 0x0000000000000044 |   |   |   | ControlCore.cpp |   | 494

</body>
</html>

**Comment by zadjii-msft** [comment]
IT'S SUPER BAD

**Comment by zadjii-msft** [comment]
Wait, this _isn't_ the UnicodeStorage one
